### PR TITLE
fix(ci): Conditional to build contracts in new CI

### PR DESCRIPTION
**Pull request by artmakh**
## What ❔

Do not build contracts if not needed in new CI

## Why ❔

Speed up CI

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

### CODE REVIEW

File: .github/workflows/new-build-core-template.yml [hunk 1]
```diff
@@ -135,17 +143,20 @@ jobs:
           done
 
       - name: init
+        if: env.BUILD_CONTRACTS == 'true'
         shell: bash
         run: |
           mkdir -p ./volumes/postgres
           docker compose up -d postgres
```
Comment (Deniallugo) [inline .github/workflows/new-build-core-template.yml:150]: we don't need it at all. let's try to remove

File: .github/workflows/new-build-core-template.yml [hunk 2]
```diff
@@ -79,36 +79,43 @@ jobs:
...
         uses: baptiste0928/cargo-install@904927dbe77864e0f2281519fe9d5bd097a220b3 # v3.1.1
         with:
           crate: cargo-nextest
 
       - name: Install sqlx-cli from crates.io
+        if: env.BUILD_CONTRACTS == 'true'
         uses: baptiste0928/cargo-install@904927dbe77864e0f2281519fe9d5bd097a220b3 # v3.1.1
         with:
           crate: sqlx-cli
           tag: 0.8.1
```
Comment (Deniallugo) [inline .github/workflows/new-build-core-template.yml:115]: Let's remove it as well

File: .github/workflows/new-build-contract-verifier-template.yml [hunk 3]
```diff
@@ -74,36 +74,43 @@ jobs:
...
         uses: baptiste0928/cargo-install@904927dbe77864e0f2281519fe9d5bd097a220b3 # v3.1.1
         with:
           crate: cargo-nextest
 
       - name: Install sqlx-cli from crates.io
+        if: env.BUILD_CONTRACTS == 'true'
         uses: baptiste0928/cargo-install@904927dbe77864e0f2281519fe9d5bd097a220b3 # v3.1.1
         with:
           crate: sqlx-cli
           tag: 0.8.1
```
Comment (Deniallugo) [inline .github/workflows/new-build-contract-verifier-template.yml:110]: redundant

File: .github/workflows/new-build-contract-verifier-template.yml [hunk 4]
```diff
@@ -130,17 +138,20 @@ jobs:
           done
 
       - name: init
+        if: env.BUILD_CONTRACTS == 'true'
         shell: bash
         run: |
           mkdir -p ./volumes/postgres
           docker compose up -d postgres
```
Comment (Deniallugo) [inline .github/workflows/new-build-contract-verifier-template.yml:146]: not required

File: .github/workflows/new-build-contract-verifier-template.yml [hunk 5]
```diff
@@ -74,36 +74,36 @@ jobs:
...
+        if: env.BUILD_CONTRACTS == 'true'
         uses: actions-rust-lang/setup-rust-toolchain@1fbea72663f6d4c03efaab13560c8a24cfd2a7cc # v1.9.0
         with:
           toolchain: nightly-2024-08-01
 
       - name: Install cargo-nextest from crates.io
+        if: env.BUILD_CONTRACTS == 'true'
         uses: baptiste0928/cargo-install@904927dbe77864e0f2281519fe9d5bd097a220b3 # v3.1.1
         with:
           crate: cargo-nextest
```
Comment (Deniallugo) [inline .github/workflows/new-build-contract-verifier-template.yml:103]: and this

File: .github/workflows/new-build-core-template.yml [hunk 6]
```diff
@@ -79,36 +79,36 @@ jobs:
...
+        if: env.BUILD_CONTRACTS == 'true'
         uses: actions-rust-lang/setup-rust-toolchain@1fbea72663f6d4c03efaab13560c8a24cfd2a7cc # v1.9.0
         with:
           toolchain: nightly-2024-08-01
 
       - name: Install cargo-nextest from crates.io
+        if: env.BUILD_CONTRACTS == 'true'
         uses: baptiste0928/cargo-install@904927dbe77864e0f2281519fe9d5bd097a220b3 # v3.1.1
         with:
           crate: cargo-nextest
```
Comment (Deniallugo) [inline .github/workflows/new-build-core-template.yml:109]: and this

**Review by yorik — APPROVED** [review]

**Pull request merged**